### ISSUE TITLE
[infra] AERA-v26.10 #709 batch64 logit-equivalence OOM harness repair1

### ISSUE THREAD

**Issue by vinceackermann2-sys** [vinceackermann2-sys/tam-research#710]
## Purpose

Fresh harness-only successor after consumed #709 failed from evaluator GPU memory pressure. Preserve the frozen v26.10 latent-depth sync-coalescing candidate byte-for-byte and repair only correctness-evaluation memory behavior so the already-preregistered two-batch optimization microbenchmark can be executed once under fresh governance.

## Exact live source

At issue creation:
- `SOURCE_MAIN=c2e9c17d7bb34d7652fc6759c80193f2d951f2ce`
- `SOURCE_TREE=ff3913de8e605aba10f87839676c9b582304233c`

Frozen v26.10 candidate remains:
- implementation blob `d8f691c198eed1fa96bcbb78a4e76cad82d18779`
- original #706 launcher blob `e23072aa83b0bc20ccd72e7567f234e9f4a931f6`
- frozen candidate commit `5cac6325fb3047f4a573d5b6e8eeeda202834c16`
- checkpoint `/vol/aera-real-language/v25-dev-seed8471`
- AERA checkpoint SHA256 `f8aa92421801e8f190247e420632be5f0c20bc5ea8bf6bdeefe06686b3a31b30`
- Transformer checkpoint SHA256 `cdd5cab4439a709468d6607d45d82081b33e876b2e40d91d4a38ba139b219dd7`

No learned/runtime candidate file may be modified by this successor.

## Consumed predecessor failure

Canonical consumed L4 trigger: `#709`
- incident classification comment `5573348027`
- precision-correction comment `5573367269`
- run `34141205064`
- job `101803372290`
- attempt `1`
- one NVIDIA L4
- bound main `c2e9c17d7bb34d7652fc6759c80193f2d951f2ce`

The run reached the authorized L4 function and failed in frozen `base._logit_equivalence` at:
`(reference.float() - candidate.float()).abs().max()`
with `torch.OutOfMemoryError`, requesting `6.14 GiB` while only `3.01 GiB` was free.

Frozen dimensions are `VOCAB_SIZE=50257`, `SEQ_LEN=512`; one batch-64 FP32 logits tensor is `6.1348876953125 GiB`, exactly localizing the failure to batch-64 full-FP32 correctness comparison.

#709 and its run/job/attempt and old #706 L4 trigger namespace are permanently consumed and MUST NOT be rerun/retried/redispatched.

No complete two-batch result/evidence was persisted. Any transient batch-8 measurements receive zero scientific/system credit.

## Exact repair mechanism

Harness-only. Preserve the original #706 evaluator and benchmark structure except for logit-equivalence memory materialization:
- require the same exact dtype/device/shape metadata equality;
- iterate over the logits along batch dimension with `LOGIT_COMPARE_BATCH_CHUNK=1`;
- for each chunk only, cast reference and candidate chunk to FP32;
- compute that chunk's absolute-difference maximum;
- apply `torch.allclose` with the same frozen `atol=1e-2`, `rtol=1e-2`, default `equal_nan=False`;
- aggregate global `max_abs = max(chunk_max_abs)` and global `allclose = AND(chunk_allclose)`;
- return the exact same logit-equivalence schema and PASS predicate as frozen `_logit_equivalence`;
- do not change logits, model state, routes, experts, latent depths, active-row counts, FICEM, writes, timing fixtures, warmups, profiler fixtures, or checkpoint bytes.

CPU/static tests must prove on representative tensors that the chunked comparator is result-identical to the frozen full comparator for pass, fail, metadata mismatch, tolerance-boundary, NaN, and max-absolute-difference cases.

The new L4 harness may use a fresh result path but must compare the exact same merged v26.9 baseline and exact frozen v26.10 candidate.

## Frozen scientific/performance gate — unchanged

Fixtures remain exactly:
- batches `(8,64)`
- token seed rule `138471 + 10000 + batch_size`
- `route_mode="hard_sparse"`
- `hard=True`
- `update_memory=True`
- CUDA BF16 autocast through the frozen model call
- 3 warmup calls
- 20 separately measured unprofiled CUDA-event timing calls per condition
- one post-timing profiler call per condition
- integrated correctness tolerances `atol=1e-2`, `rtol=1e-2`
- candidate median latency improvement `>=5%` at batch 8 AND batch 64
- at least `20` fewer `cudaStreamSynchronize` calls at batch 8 AND batch 64
- every route/expert/depth/active-row/state/schema/checkpoint/sparsity gate remains mandatory.

PASS authorizes only a separately preregistered fresh full frozen Transformer-relative E2E systems gate. It does not earn systems PASS itself. FAIL authorizes no E2E rerun.

## Fresh governance

Use only fresh successor namespaces, planned as:
- branch `infra/aera-v26-10-issue710-memory-safe-harness-repair1`
- preauth prefix `[aera-v26-10-issue710-memory-safe-harness-preauth]`
- L4 prefix `[aera-v26-10-issue710-memory-safe-harness-l4]`
- result path `/vol/aera-v26/issue710-v26-10-memory-safe-harness/result.json`

Before any branch/ref move:
1. create additive successor-specific launcher/workflow/CPU-test blobs as detached Git objects;
2. freeze exact source SHA/tree, #709 incident lineage, old/new blobs, checkpoint hashes, result path, prefixes, thresholds, detached tree and detached commit in one comment;
3. only then create the named infra branch at exactly the frozen detached commit;
4. open one immutable PR and consume exactly one original full-repository CPU CI attempt;
5. never rerun/retry/redispatch a consumed attempt.

After CPU-green merge: fresh exact-main audit, one fresh zero-GPU preauthorization trigger, separate exact-main L4 authorization, then exactly one fresh one-L4 trigger. No Modal/GPU execution is authorized before those gates.

## Authority ceiling

This issue is evaluator-infrastructure only. It authorizes no candidate optimization change, no systems PASS, no architecture freeze, no S2, no fresh scientific seed, no independent replication credit, no 100M scaling, and no breakthrough claim.

**Comment by vinceackermann2-sys** [comment]
## #710 pre-implementation memory-safe harness repair1 freeze

This freezes the harness-only successor after consumed #709 and before any #710 branch/ref move. No Modal/GPU execution, optimization result, full E2E systems run, systems PASS, architecture freeze, S2, fresh scientific seed, replication, scaling, or breakthrough authority is created by this comment.

SOURCE_MAIN=c2e9c17d7bb34d7652fc6759c80193f2d951f2ce
SOURCE_TREE=ff3913de8e605aba10f87839676c9b582304233c
BRANCH=infra/aera-v26-10-issue710-memory-safe-harness-repair1
PREAUTH_PREFIX=[aera-v26-10-issue710-memory-safe-harness-preauth]
L4_PREFIX=[aera-v26-10-issue710-memory-safe-harness-l4]
RESULT_PATH=/vol/aera-v26/issue710-v26-10-memory-safe-harness/result.json

CONSUMED_TRIGGER=709
CONSUMED_RUN=34141205064
CONSUMED_JOB=101803372290
CONSUMED_ATTEMPT=1
CONSUMED_FAILURE_COMMENT=5573348027
CONSUMED_PRECISION_COMMENT=5573367269

CHECKPOINT_AERA_SHA256=f8aa92421801e8f190247e420632be5f0c20bc5ea8bf6bdeefe06686b3a31b30
CHECKPOINT_TRANSFORMER_SHA256=cdd5cab4439a709468d6607d45d82081b33e876b2e40d91d4a38ba139b219dd7
ISSUE706_LAUNCHER_BLOB=e23072aa83b0bc20ccd72e7567f234e9f4a931f6
V26_10_IMPL_BLOB=d8f691c198eed1fa96bcbb78a4e76cad82d18779
ISSUE665_LAUNCHER_BLOB=72f27391ff2f0a7bff8d4532f307ddc4869cf494
V25_1_NOHOST_BLOB=237e5615cf32f644e8675808a6b3e9adaf04fb23
V26_RUNTIME_BLOB=268644ac4edee15a4cc4e29d3fed7f61eeb3caa7
V26_9_SYSTEMS_BLOB=512572340cc09e2e7ad6729712258c12cb377ef2
BASE_SYSTEMS_BLOB=c9731cae7e386f09b2a190b045532591c4fa00be
TRIAGE_BLOB=e5ce8cdda0777dce97816e2640f4492803a6b191
V26_9_BACKEND_BLOB=b81cc209f5d95abbe1fb8bd620c78e87c067bc19

LOGIT_COMPARE_BATCH_CHUNK=1
BATCHES=8,64
TOKEN_SEED_RULE=138471+10000+batch_size
ROUTE_MODE=hard_sparse
HARD=True
UPDATE_MEMORY=True
CUDA_AUTOCAST=BF16
WARMUP_CALLS=3
TIMED_CALLS_PER_CONDITION=20
PROFILE_CALLS_PER_CONDITION=1
MIN_LATENCY_IMPROVEMENT=0.05
MIN_STREAM_SYNCHRONIZE_REDUCTION=20
INTEGRATED_ATOL=0.01
INTEGRATED_RTOL=0.01

LAUNCHER_BLOB=b885250753ea169cd89dd7a978bb3647fd261fe8
WORKFLOW_BLOB=53e1e22c0b421c51ef6299e777132d7a21dc771d
CPU_TEST_BLOB=5ce587e83b388a1bda3494d1eff54f4535b483dc
FROZEN_TREE=fd171423b5878c9082e689d7d83da9f8c7c81569
FROZEN_COMMIT=e0a28c40a6336a09cd407245ef1bf24e5f90d131

Exact diff from SOURCE_MAIN: three additive harness files only, 920 additions, 0 deletions. The frozen v26.10 implementation and original #706 launcher are unchanged.

Memory-safe correctness rule: compare logits in batch chunks of exactly one; cast only corresponding slices to FP32; aggregate global max-absolute error with `torch.maximum` so NaN propagation matches the inherited full-tensor `.max()`; aggregate allclose by logical AND using exact inherited `atol=rtol=1e-2`, default `equal_nan=False`; require exact dtype/device/shape metadata. After all correctness predicates are computed, dead correctness outputs are released before warmups/timing; the contents/order/count of every warmup, unprofiled timed call and post-timing profiler call remain the frozen #706 fixtures.

The CPU contract must prove result identity with the inherited `_logit_equivalence` on exact pass, tolerance pass/fail, dtype metadata mismatch, NaN, and a maximum error located in a later chunk.

Only after this freeze may the named branch be created at exactly FROZEN_COMMIT, followed by one immutable PR and one original full-repository CPU CI attempt. Never rerun/retry/redispatch a consumed attempt.

**Comment by vinceackermann2-sys** [comment]
## #710 sole L4 memory-safe optimization microbenchmark authorization

Authorize main: `8371d9ab3988df33c7b92b529016618ebf60f9ee`

Authorize exactly one fresh L4 optimization microbenchmark trigger under frozen prefix `[aera-v26-10-issue710-memory-safe-harness-l4]`, and nothing else.

Successful required preauthorization:
- trigger issue `#714`
- evidence comment `5573639026`
- workflow run `34145132878`
- job `101815411355`
- attempt `1`
- `result_absent=true`
- `gpu_used=false`
- `model_constructed=false`
- `new_measurement_performed=false`

Frozen harness boundary:
- merged main `8371d9ab3988df33c7b92b529016618ebf60f9ee`
- merged/frozen tree `fd171423b5878c9082e689d7d83da9f8c7c81569`
- frozen harness commit `e0a28c40a6336a09cd407245ef1bf24e5f90d131`
- memory-safe launcher blob `b885250753ea169cd89dd7a978bb3647fd261fe8`
- workflow blob `53e1e22c0b421c51ef6299e777132d7a21dc771d`
- CPU-test blob `5ce587e83b388a1bda3494d1eff54f4535b483dc`
- frozen v26.10 implementation blob `d8f691c198eed1fa96bcbb78a4e76cad82d18779` unchanged
- original #706 launcher blob `e23072aa83b0bc20ccd72e7567f234e9f4a931f6` unchanged
- logit comparison batch chunk `1`
- batches `(8,64)`, token seed rule `138471 + 10000 + batch_size`, hard-sparse routing, `hard=True`, `update_memory=True`, CUDA BF16 autocast
- 3 warmups, 20 separately measured unprofiled CUDA-event timing calls per condition, one post-timing profiler call
- correctness tolerances `atol=1e-2`, `rtol=1e-2`
- latency gate `>=5%` improvement at both batches
- mechanism gate `>=20` fewer `cudaStreamSynchronize` calls at both batches
- every frozen route/expert/depth/active-row/state/schema/checkpoint/sparsity gate remains mandatory before performance credit.

This authorization permits only the one #710 L4 memory-safe optimization microbenchmark. A PASS may authorize only a separately preregistered fresh full frozen Transformer-relative E2E systems gate; it does not itself earn systems PASS. No architecture freeze, S2, fresh scientific seed, independent replication credit, 100M scaling, or breakthrough claim is authorized. The L4 trigger and its Actions attempt are consumed regardless of outcome; never rerun/retry/redispatch.
